hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/UpdateTokenExpiryInfoPrecompile.java

```diff
@@ -0,0 +1,112 @@
+
+import com.hedera.services.context.SideEffectsTracker;
+import com.hedera.services.contracts.sources.EvmSigsVerifier;
+import com.hedera.services.ledger.accounts.ContractAliases;
+import com.hedera.services.store.contracts.WorldLedgers;
+import com.hedera.services.store.contracts.precompile.InfrastructureFactory;
+import com.hedera.services.store.contracts.precompile.SyntheticTxnFactory;
+import com.hedera.services.store.contracts.precompile.TokenUpdateLogic;
+import com.hedera.services.store.contracts.precompile.codec.DecodingFacade;
+import com.hedera.services.store.contracts.precompile.codec.TokenUpdateExpiryInfoWrapper;
+import com.hedera.services.store.contracts.precompile.utils.KeyActivationUtils;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hedera.services.store.models.Id;
+import com.hedera.services.store.tokens.HederaTokenStore;
+import com.hederahashgraph.api.proto.java.Timestamp;
+import com.hederahashgraph.api.proto.java.TransactionBody.Builder;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+
+public class UpdateTokenExpiryInfoPrecompile extends AbstractWritePrecompile {
+    private HederaTokenStore hederaTokenStore;
+    private TokenUpdateExpiryInfoWrapper updateExpiryInfoOp;
+    private final EvmSigsVerifier sigsVerifier;
+    private final ContractAliases aliases;
+
+    public UpdateTokenExpiryInfoPrecompile(
+            WorldLedgers ledgers,
+            ContractAliases aliases,
+            DecodingFacade decoder,
+            EvmSigsVerifier sigsVerifier,
+            SideEffectsTracker sideEffects,
+            SyntheticTxnFactory syntheticTxnFactory,
+            InfrastructureFactory infrastructureFactory,
+            PrecompilePricingUtils pricingUtils) {
+        super(
+                ledgers,
+                decoder,
+                sideEffects,
+                syntheticTxnFactory,
+                infrastructureFactory,
+                pricingUtils);
+        this.aliases = aliases;
+        this.sigsVerifier = sigsVerifier;
+    }
+
+    @Override
+    public Builder body(Bytes input, UnaryOperator<byte[]> aliasResolver) {
+        updateExpiryInfoOp = decoder.decodeUpdateTokenExpiryInfo(input, aliasResolver);
+        transactionBody = syntheticTxnFactory.createTokenUpdateExpiryInfo(updateExpiryInfoOp);
+        initializeHederaTokenStore();
+        return transactionBody;
+    }
+
+    @Override
+    public long getMinimumFeeInTinybars(Timestamp consensusTime) {
+        return pricingUtils.getMinimumPriceInTinybars(UPDATE, consensusTime);
+    }
+
+    @Override
+    public void run(MessageFrame frame) {
+        Objects.requireNonNull(updateExpiryInfoOp);
+        /* --- Check required signatures --- */
+        validateFalse(updateExpiryInfoOp.tokenID() == MISSING_TOKEN, INVALID_TOKEN_ID);
```

Comment:
Probably it would be more clear if you use:
`validateTrue(updateExpiryInfoOp.tokenID() !=null, INVALID_TOKEN_ID)`